CSharpSourceCode/Battle/Dismemberment/Dismemberment.cs

```diff
@@ -0,0 +1,158 @@
+
+            foreach (Mesh mesh in victim.AgentVisuals.GetSkeleton().GetAllMeshes())
+            {
+                if ((mesh.Name.Contains("head") || mesh.Name.Contains("_hat_")) && !mesh.Name.Contains("lod"))
+                {
+                    Mesh childMesh = mesh.GetBaseMesh().CreateCopy();
+                    var child = GameEntity.CreateEmpty(Mission.Current.Scene, true);
+                    childMesh.SetLocalFrame(headLocalFrame);
+                    child.AddMesh(childMesh);
+                    head.AddChild(child);
+                }
+            }
+
+            String[] meshNames = { "hair", "beard", "eyebrow", "_cap_", "helmet" };
+            foreach (String name in meshNames)
+            {
+                Mesh mesh = victim.AgentVisuals.GetSkeleton().GetAllMeshes().FirstOrDefault(m => m.Name.Contains(name));
+                if (mesh != default(Mesh))
+                {
+                    Mesh childMesh = mesh.GetBaseMesh().CreateCopy();
+                    var child = GameEntity.CreateEmpty(Mission.Current.Scene, true);
+                    childMesh.SetLocalFrame(headLocalFrame);
+                    child.AddMesh(childMesh);
+                    head.AddChild(child);
+                }
+            }
+            return head;
+        }
+        private static void AddPhysicsFromCollision(GameEntity head, AttackCollisionData attackCollision)
+        {
+            Vec3 blowDir = attackCollision.WeaponBlowDir;
+            Vec3 velocityVec = new Vec3(blowDir.X * 3, blowDir.Y * 3);
+            Vec3 angularVec = new Vec3(velocityVec.X * 3, velocityVec.Y * 3);
+            head.AddPhysics(1f, head.CenterOfMass, head.GetBodyShape(), velocityVec, angularVec, PhysicsMaterial.GetFromName("flesh"), false, -1);
+        }
+        private static void CreateBloodBurst(Agent victim)
+        {
+            MatrixFrame boneEntitialFrameWithIndex = victim.AgentVisuals.GetSkeleton().GetBoneEntitialFrameWithIndex((byte)victim.BoneMappingArray[HumanBone.Head]);
+            Vec3 vec = victim.AgentVisuals.GetGlobalFrame().TransformToParent(boneEntitialFrameWithIndex.origin);
+            victim.CreateBloodBurstAtLimb(13, ref vec, 0.5f + MBRandom.RandomFloat * 0.5f);
+        }
+
+        public static Agent SpawnAgent(Agent agent)
+        {
+            AgentBuildData agentBuildData = new AgentBuildData(agent.Character);
+            agentBuildData.NoHorses(true);
+            agentBuildData.NoWeapons(true);
+            agentBuildData.NoArmor(false);
+            agentBuildData.Team(Mission.Current.PlayerEnemyTeam);
+            agentBuildData.TroopOrigin(agent.Origin);
+
+            MatrixFrame frame = default(MatrixFrame);
+            frame.origin = agent.Position;
+            frame.rotation = agent.Frame.rotation;
+            agentBuildData.InitialFrame(frame);
+
+            return Mission.Current.SpawnAgent(agentBuildData, false, 0);
+        }
```

Comment:
This could be moved into a Utilities class called something like `TowDebug` if we want to keep it. I don't think it fits to be in a dismemberment-specific class but could be useful in the future.

Reply:
agree use other Debug Functions , instead of creating those.